common/beerocks/bcl/unit_tests/wireless_utils_test.cpp

```diff
@@ -13,120 +13,265 @@
+                    if (operating_class.bandwidth != bandwidth) {
+                        return false;
+                    }
+                    return operating_class.channels.end() != operating_class.channels.find(channel);
+                });
+            if (it == table.end()) {
+                beerocks::message::sWifiChannel ch;
+                ch.channel           = get_starting_channel(channel, bandwidth);
+                ch.channel_bandwidth = bandwidth;
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(52, 40), 119);
-    EXPECT_EQ(get_operating_class_by_channel_helper(60, 40), 119);
+                result.emplace_back(std::make_tuple(ch, 0));
+            }
+        }
+    } while (channel++ < UINT8_MAX);
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(56, 40), 120);
-    EXPECT_EQ(get_operating_class_by_channel_helper(64, 40), 120);
+    return result;
+}
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(100, 20), 121);
-    EXPECT_EQ(get_operating_class_by_channel_helper(144, 20), 121);
+/**
+ * @brief Converts a channel number to its string representation.
+ *
+ * Helper function to build the channel part of the name of a value-parameterized test parameter.
+ *
+ * @param value Channel number.
+ * @return Channel number as a string.
+ */
+std::string channel_to_string(uint8_t value) { return "channel_" + std::to_string(value); }
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(100, 40), 122);
-    EXPECT_EQ(get_operating_class_by_channel_helper(140, 40), 122);
+/**
+ * @brief Converts a channel bandwidth to its string representation.
+ *
+ * Helper function to build the bandwidth part of the name of a value-parameterized test parameter.
+ *
+ * @param value Channel bandwidth.
+ * @return Channel bandwidth as a string.
+ */
+std::string bandwidth_to_string(uint8_t value)
+{
+    // clang-format off
+    const std::map<beerocks::eWiFiBandwidth, std::string> bandwidths{
+        {beerocks::BANDWIDTH_UNKNOWN, "0"},
+		{beerocks::BANDWIDTH_20, "20"},
+        {beerocks::BANDWIDTH_40, "40"},
+		{beerocks::BANDWIDTH_80, "80"},
+        {beerocks::BANDWIDTH_80_80, "80_80"},
+		{beerocks::BANDWIDTH_160, "160"}
+    };
+    // clang-format on
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(104, 40), 123);
-    EXPECT_EQ(get_operating_class_by_channel_helper(144, 40), 123);
+    std::string result = "bandwidth_";
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(149, 20), 124);
-    EXPECT_EQ(get_operating_class_by_channel_helper(161, 20), 124);
+    auto bandwidth = static_cast<beerocks::eWiFiBandwidth>(value);
 
-    EXPECT_EQ(get_operating_class_by_channel_helper(165, 20), 125);
-    EXPECT_EQ(get_operating_class_by_channel_helper(169, 20), 125);
+    const auto &it = bandwidths.find(bandwidth);
+    if (bandwidths.end() == it) {
+        result += "invalid_" + std::to_string(value);
+    } else {
+        result += it->second;
+    }
```

Comment:
(nitpick)
I would have:
- used the full strings directly, rather than += (i.e. `{beerocks::BANDWIDTH_20, "bandwidth_20"},`);
- used a switch rather than a lookup table.

Reply:
Some static analyzers include a rule by which `switch` statements should not have too many `case` clauses: "When switch statements have large sets of case clauses, it is usually an attempt to map two sets of data. A real map structure would be more readable and maintainable, and should be used instead."

For the sake of curiosity, why do you prefer a switch rather than a lookup table?

Reply:
Because I think a switch is more readable and maintainable :-)

Many lint tools and even gcc with -Wswitch enabled (which is part of -Wall) will flag when an enum value is missing in a switch. AFAIK, the same thing doesn't exist in a look-up table.

In terms of readability, I think a switch is pretty readable. It would be nicer if there was a switch expression though... But I guess indeed a look-up table is more readable, at least if it's easily representable as a look-up table (which is the case here).